src/redux/reducers/AvailRequestsReducer.js

```diff
@@ -0,0 +1,37 @@
+import { GET_REQUESTS_FAILURE, GET_REQUESTS_SUCCESS, GET_REQUESTS } from '../types/requestsTypes';
+
+const initialState = {
+	loading: 'block',
+	requests: [],
+	error: '',
+	open: false
+};
+
+const reducer = (state = initialState, action) => {
+	switch (action.type) {
+		case GET_REQUESTS:
+			return {
+				...state,
+				loading: 'block',
+				open: false
+			};
+		case GET_REQUESTS_SUCCESS:
+			return {
+				loading: 'none',
+				requests: action.payload,
+				error: '',
+				open: false
+			};
+		case GET_REQUESTS_FAILURE:
+			return {
+				loading: 'none',
+				requests: [],
+				error: action.payload,
+				open: true
+			};
```

Comment:
is this destructuring or spreading?

Reply:
Exactly, that's what I meant

Reply:
> is this destructuring or spreading?

Exactly, that's what I meant

Reply:
Then the suggestion your making can't be applied here.

Suggestion 
```
	case GET_REQUESTS_SUCCESS:
			return {
			        ...state,
				loading: 'none',
				requests: action.payload,
			};
```

on the success, we have to put loading: 'none', and make sure that we clear the error value 	error: '',

so spreading ...state won't be applicable since all values of that state have to change.


this would be more useful if we were to be maybe using next not dispatch and we want to keep some values as they were before.

https://stackoverflow.com/questions/36160457/whats-the-difference-between-dispatch-and-next-in-redux-middleware